generic/build

```diff
@@ -365,8 +365,8 @@ solaris_common() {
 main() {
 	export CFLAGS="${TARGET_CFLAGS} ${EXTRA_TARGET_CFLAGS}"
 	export LDFLAGS="${TARGET_LDFLAGS} ${EXTRA_TARGET_LDFLAGS}"
-	download
 	create_layout
+	download
 	extract
```

Comment:
I would keep this part in a separate commit, because it's entirely unrelated to the curl stuff. Maybe in a commit called "Create layout before downloading dependency packages"?